test/iron-request.html

```diff
@@ -84,6 +84,12 @@
           async: true,
           timeout: 100
         };
+
+        asynchronousUndefSuccessfulRequestOptions = {
```

Comment:
nit: given that the only case where a request won't be async is when you pass `options.async = false`, WDYT of just removing `async: true` from `asynchronousSuccessfulRequestOptions` instead of having `asynchronousUndefSuccessfulRequestOptions`? That way this test won't be opinionated on other values of `options.async` other than `false`.

Reply:
I believe that it is important to specifically test a common expected input such as `true`. Also if anything I believe that async should be false if the value is falsey except for `undefined`